### PR TITLE
[PM-11404] Account Management: Prevent a verified user from purging their vault

### PR DESCRIPTION
**Pull request by r-tome**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11404

## 📔 Objective

If the Account Deprovisioning feature flag is enabled and the user is managed by an organization then hide the `Purge vault` button.

Server PR: https://github.com/bitwarden/server/pull/4853

## 📸 Screenshots

Feature flag enabled and account managed:
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by coroiu — DISMISSED** [review]
approving platform owned code

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
LGTM! Appreciate the docs on `managesActiveUser` as well!

**Review by jrmccannon — APPROVED** [review]

File: libs/common/src/admin-console/models/domain/organization.ts [hunk 1]
```diff
@@ -73,6 +73,12 @@ export class Organization {
    * Refers to the ability for an owner/admin to access all collection items, regardless of assigned collections
    */
   allowAdminAccessToAllCollectionItems: boolean;
+  /**
+   * Indicates if this organization manages the user.
+   * A user is considered managed by an organization if their email domain
+   * matches one of the verified domains of that organization, and the user is a member of it.
+   */
+  userIsManagedByOrganization: boolean;
```
Comment (coroiu) [inline libs/common/src/admin-console/models/domain/organization.ts:81]: praise: appreciate the docs!

question: What happens if Google creates an organization and verifies `gmail.com`?

**Review by coroiu — APPROVED** [review]
I don't think Platform owns any files in this PR anymore? You should be able to merge without my approval if you make any additional changes

File: libs/common/src/admin-console/models/domain/organization.ts (hunk 1 above)
Comment (r-tome) [reply libs/common/src/admin-console/models/domain/organization.ts:81]: > I don't think Platform owns any files in this PR anymore?

You're right, apologies for not realising that!

> question: What happens if Google creates an organization and verifies gmail.com?

They'll only be able to manage users that are members of their Bitwarden Organization

**Review by jrmccannon — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**